scripts/manila-share-sync.py

```diff
@@ -193,6 +199,66 @@ def sync_share_size(self, shares, dry_run=True):
                         self.set_share_size(share_id, vsize)
                         self.MANILA_SYNC_SHARE_SIZE_COUNTER.inc()
 
+    def sync_share_state(self, shares, dry_run=True):
+        """ Deal with share in stuck states for more than 15 minutes """
+        msg = "ManilaSyncShareState: share=%s, instance=%s status=%s"
+        msg_dry_run = "Dry run: " + msg
+        for share in shares:
+            share_status = share['status']
+            if share_status not in ['creating', 'error_deleting', 'deleting']:
+                continue
+
+            share_id = share['id']
+            instances = self._query_share_instances(share_id)
+            if len(instances) == 0:
+                continue
+            elif len(instances) == 1:
+                instance = instances[0]
+                instance_updated_at = instance['updated_at']
+                if instance_updated_at is not None:
+                    if is_utcts_recent(instance_updated_at, 900):
+                        continue
+                else:
+                    continue
+
+                instance_id = instance['id']
+                instance_status = instance['status']
+                if dry_run:
+                    log.info(msg_dry_run, share_id, instance_id, instance_status)
+                    continue
+                log.info(msg, share_id, instance_id, instance_status)
+                if share_status == 'error_deleting':
+                    self.share_force_delete(share_id)
+                    continue
+                self.share_reset_state(share_id, 'error')
+                if share_status == 'deleting':
+                    self.share_delete(share_id)
+            else:
+                for instance in instances:
+                    instance_updated_at = instance['updated_at']
+                    if instance_updated_at is not None:
+                        if is_utcts_recent(instance_updated_at, 900):
+                            continue
+                    else:
+                        continue
+
+                    if instance.get('replica_state', None) in ['active', None]:
```

Comment:
get() function returns None when the key is not present in the `instance` dict. So no need to add None as default value.